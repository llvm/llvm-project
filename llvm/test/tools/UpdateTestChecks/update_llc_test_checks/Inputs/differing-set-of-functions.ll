; RUN: sed 's/FN/foo/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKA %s
; RUN: sed 's/FN/foo/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKB %s
; RUN: sed 's/FN/bar/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKC %s

define i32 @FN() {
  ret i32 1
}

define i32 @common() {
  ret i32 100
}
