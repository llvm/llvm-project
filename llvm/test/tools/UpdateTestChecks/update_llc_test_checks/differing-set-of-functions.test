# REQUIRES: riscv-registered-target

# RUN: cp -f %S/Inputs/differing-set-of-functions.ll %t.ll
# RUN: %update_llc_test_checks --include-generated-funcs %t.ll 2>&1 | FileCheck --allow-empty %s
# RUN: diff -u %S/Inputs/differing-set-of-functions.ll.expected %t.ll

# We shouldn't print the warning for clashing CHECK prefixes in the case that
# we're trying to handle a function that is only present for some RUN lines.
# Better warning behaviour than this might be possible.

# CHECK-NOT: WARNING
