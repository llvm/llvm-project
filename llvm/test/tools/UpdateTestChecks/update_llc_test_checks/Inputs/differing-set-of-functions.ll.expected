; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --include-generated-funcs
; RUN: sed 's/FN/foo/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKA %s
; RUN: sed 's/FN/foo/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKB %s
; RUN: sed 's/FN/bar/g' %s | llc -mtriple=riscv32 \
; RUN:   | FileCheck -check-prefixes=CHECK,CHECKC %s

define i32 @FN() {
  ret i32 1
}

define i32 @common() {
  ret i32 100
}
; CHECK-LABEL: foo:
; CHECK:       # %bb.0:
; CHECK-NEXT:    li a0, 1
; CHECK-NEXT:    ret
;
; CHECK-LABEL: common:
; CHECK:       # %bb.0:
; CHECK-NEXT:    li a0, 100
; CHECK-NEXT:    ret
;
; CHECKA-LABEL: foo:
; CHECKA:       # %bb.0:
; CHECKA-NEXT:    li a0, 1
; CHECKA-NEXT:    ret
;
; CHECKA-LABEL: common:
; CHECKA:       # %bb.0:
; CHECKA-NEXT:    li a0, 100
; CHECKA-NEXT:    ret
;
; CHECKB-LABEL: foo:
; CHECKB:       # %bb.0:
; CHECKB-NEXT:    li a0, 1
; CHECKB-NEXT:    ret
;
; CHECKB-LABEL: common:
; CHECKB:       # %bb.0:
; CHECKB-NEXT:    li a0, 100
; CHECKB-NEXT:    ret
;
; CHECKC-LABEL: bar:
; CHECKC:       # %bb.0:
; CHECKC-NEXT:    li a0, 1
; CHECKC-NEXT:    ret
;
; CHECKC-LABEL: common:
; CHECKC:       # %bb.0:
; CHECKC-NEXT:    li a0, 100
; CHECKC-NEXT:    ret
